toolbox/include/base/ToolboxSingleton.h

```diff
@@ -0,0 +1,142 @@
+#ifndef WBT_ToolboxSingleton_H
+#define WBT_ToolboxSingleton_H
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <unordered_map>
+#include "Configuration.h"
+
+namespace wbt {
+    class ToolboxSingleton;
+    class RobotInterface;
+}
+
+namespace iDynTree {
+    class KinDynComputations;
+}
+
+// TODO: check class doxygen macro
+/**
+ * \class ToolboxSingleton ToolboxSingleton.h
+ *
+ * This class handles and contains configuration and devices used by all the blocks
+ * allocated by a running model. It is possible obtaining a singleton of this class
+ * by calling WBToolbox::sharedInstance.
+ *
+ * @see wbt::RobotInterface
+ * @see wbt::Configuration
+ * @see wbt::yarpDevices
+ *
+ */
+class wbt::ToolboxSingleton { // TODO: wbt::ToolboxSingleton
+private:
+    /// Object that stores all the configurations labelled by the name of the Simulink Block's
+    /// name.
+    /// @see wbt::RobotInterface
+    std::unordered_map<std::string, std::shared_ptr<wbt::RobotInterface>> m_interfaces;
+
+public:
+
+    // CONSTRUCTOR / DESTRUCTOR
+    // ========================
+
+    ToolboxSingleton();
+    ToolboxSingleton(const ToolboxSingleton&);
```

Comment:
If it is a singleton both copy and assignment should be prohibited

Reply:
Used `= delete`